Library/Homebrew/cmd/pull.rb

```diff
@@ -128,8 +128,12 @@ def pull
         end
       end
 
-      unless ARGV.include? "--bottle"
-        changed_formulae.each do |f|
+      fetch_bottle = false
+      changed_formulae.each do |f|
+        if ARGV.include? "--bottle"
+          fetch_bottle ||= !!f.bottle
```

Comment:
Since it's explicitly set to false, won't these conditional assignments always be ignored?